lib/ecl/ecl_sum_file_data.cpp

```diff
@@ -0,0 +1,736 @@
+   average of the M2 and M3 values. Whether a variable in question is
+   interpreted as a 'rate' is effectively determined by the
+   ecl_smspec_set_rate_var() function in ecl_smspec.c.
+
+
+
+   Indexing and _get() versus _iget()
+   ----------------------------------
+   As already mentionded the set of ministeps is not necessarrily a
+   continous series, we can easily have a series of ministeps with
+   "holes" in it, and the series can also start on a non-zero
+   value. Internally all the ministeps are stored in a dense, zero
+   offset vector instance; and we must be able to translate back and
+   forth between ministep_nr and internal index.
+
+   Partly due to EnKF heritage the MINISTEP nr has been the main
+   method to access the time dimension of the data, i.e. all the
+   functions like ecl_sum_get_general_var() expect the time direction
+   to be given as a ministep; however it is also possible to get the
+   data by giving an internal (not that internal ...) index. In
+   ecl_sum_data.c the latter functions have _iget():
+
+
+      ecl_sum_data_get_xxx : Expects the time direction given as a ministep_nr.
+      ecl_sum_data_iget_xxx: Expects the time direction given as an internal index.
+
+*/
+
+
+namespace ecl {
+
+
+ecl_sum_file_data::ecl_sum_file_data(const ecl_smspec_type * smspec) :
+  ecl_smspec( smspec )
+{
+  data = vector_alloc_new();
+  clear_index( );
+
+}
+
+ecl_sum_file_data::~ecl_sum_file_data() {
+  vector_free( data );
+}
+
+
+int ecl_sum_file_data::get_length() const {
+  return vector_get_size( data );
+}
+
+
+int ecl_sum_file_data::length_before(time_t end_time) const {
+  int offset = 0;
+  while (true) {
+    time_t itime = this->iget_sim_time(offset);
+    if (itime >= end_time)
+      return offset;
+
+    offset += 1;
+    if (offset == this->get_length())
+      return offset;
+  }
+}
+
+
+int ecl_sum_file_data::report_before(time_t end_time) const {
+  if (end_time < this->first_report())
+    throw std::invalid_argument("time argument before first report step");
+
+  int r = this->first_report();
+  int last_report = this->last_report();
+  while (true) {
+    if (r == last_report)
+      return last_report;
+
+    auto next_range = this->report_map[r + 1];
+    if (this->iget_sim_time(next_range.first) > end_time)
+      return r;
+
+    r += 1;
+    printf("%d\n",r);
+  }
+}
+
+
+int  ecl_sum_file_data::first_report() const{
+  return first_report_step;
+}
+
+int ecl_sum_file_data::last_report() const {
+  return last_report_step;
+}
+
+
+
+time_t ecl_sum_file_data::get_data_start() const {
+  return this->time_range.first;
+}
+
+
+time_t ecl_sum_file_data::get_sim_end() const {
+  return this->time_range.second;
+}
+
+time_t ecl_sum_file_data::iget_sim_time(int time_index) const {
+  const ecl_sum_tstep_type * ministep = iget_ministep( time_index  );
+  return ecl_sum_tstep_get_sim_time(ministep);
+}
+
+double ecl_sum_file_data::iget_seconds(int time_index) const {
+  throw;
```

Comment:
Is this a mistake, or should it actually be like that?  If it's intended, please add a comment.

Reply:
Hmm - clearly mistake.